### PR TITLE
Add knative source example for Salesforce

### PR DESCRIPTION
**Pull request by igarashitm**
Note: This example requires Apache Camel 3.4.0 or newer due to a bug in camel-salesforce - https://issues.apache.org/jira/browse/CAMEL-15063

### CODE REVIEW

**Comment by lburgazzoli** [comment]
looks like the salesforce-consumer.yaml is missing

**Comment by igarashitm** [comment]
Ugh thanks for catching it - actually changed to SalesforceConsumer.java. Fixed.

File: 10-knative-source-salesforce/salesforce-source.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name: camel-salesforce-source
+spec:
+  source:
+    integration:
+      configuration:
+        - type: secret
+          value: salesforce
+    flow:
+      from:
+        uri: salesforce:camelTestTopic
```
Comment (lburgazzoli) [inline 10-knative-source-salesforce/salesforce-source.yaml:16]: maybe this should be `salesforce:{{salesforce.topic}}` to make it clear it is a parameter the user should set ?

Comment (igarashitm) [reply 10-knative-source-salesforce/salesforce-source.yaml:16]: I tried it but the property substitution doesn't seem to happen in uri, it's sending `{{salesforce.topic}}` as it is and get invalid character error.

Comment (lburgazzoli) [reply 10-knative-source-salesforce/salesforce-source.yaml:16]: even if quoted ? mind opening an issue on camel k runtime if it does not work ?

Comment (igarashitm) [reply 10-knative-source-salesforce/salesforce-source.yaml:16]: OK filed it - apache/camel-k-runtime#392

**Comment by lburgazzoli** [comment]
@igarashitm mind amending the readme to point to the exact knative version as done in https://github.com/apache/camel-k-examples/pull/13 ?

**Comment by igarashitm** [comment]
done!

**Comment by igarashitm** [comment]
Replaying with runtime 1.5.0-SNAPSHOT - do we want to use the property substitution in from uri if it works? maybe with a comment that it requires runtime 1.5.0 or newer.

**Comment by lburgazzoli** [comment]
I'd go for the example as it is, we can work on it once we have a new release of the runtime

**Comment by igarashitm** [comment]
Verified it works with 1.5.0-SNAPSHOT - I'll keep it until next release.

**Pull request merged**